Updated Python SDK version and added property management commands

jjcollinge opened the pull request:
This PR updates the azure-servicefabric Python SDK to v6.0.2 and exposes a subset of the [Property Management APIs](https://docs.microsoft.com/en-us/rest/api/servicefabric/sfclient-index-property-management) via standard commands.
I've only added commands relevant to modifying properties for existing Service Fabric names i.e. 'GettingStartedApplication/WebService' rather than the full name/property API surface. This works for our use case (adding app/service metadata) but appreciate you may want the commands to map to a more generalised API surface. 
Example usage can be seen below:
![sfctl](https://user-images.githubusercontent.com/7241190/32266217-66300b5a-bede-11e7-964e-71a3ca4a179a.PNG)

samedder commented on src/setup.py:
```diff
@@ -46,7 +46,7 @@ def read(fname):
         'msrest>=0.4.4',
         'requests',
-        'azure-servicefabric==6.0',
+        'azure-servicefabric==6.0.2',
```
CI is not passing

jjcollinge replied:
I've ran the travis script locally in a virtualenv and can repeat the failure. `pip install adal` will resolve the issue - do you want me to add 'adal' to the requirements.txt or should we fix 6.0.2 dependency graph to ensure the package is pulled in?

samedder replied:
If it's a dependency of version 6.0.2 of `azure-servicefabric` then the requirements for the package must be updated. I don't think your changes here require adal explicitly.

jjcollinge replied:
It seems that between version 6.0.0 and 6.0.1 of azure-servicefabric that the msrestazure module is not being pulled in (although it still remains referenced in the requirements.txt) and therefore adal (which is a dependency of msrestazure) is not pulled in either. There is an explicit reference to 'adal' in src/sfctl/auth.py therefore I think it makes sense to put it in sfctl's requirements.txt rather than rely on an external package's dependency - I'm no expert in python dependencies so happy to take your advice on this.

samedder replied:
Yeah it's fine to add it as a dependency for sfctl. I must have missed that whoever authored the auth changes didn't add that package.

samedder commented on src/sfctl/commands.py:
```diff
@@ -203,6 +203,12 @@ def load_command_table(self, args): #pylint: disable=too-many-statements
                 group.command('command', 'invoke_infrastructure_command')
```
Add help text for the property group now

jjcollinge replied:
Is an additional help file required for a standard command group?

samedder replied:
You added a group of commands: `sfctl property`. In the same way there's documentation on `sfctl application`. Take a look at how the help is defined for that group.

jjcollinge replied:
Ok will do, thanks.

samedder commented on src/sfctl/commands.py:
```diff
@@ -203,6 +203,12 @@ def load_command_table(self, args): #pylint: disable=too-many-statements
                 group.command('command', 'invoke_infrastructure_command')
                 group.command('query', 'invoke_infrastructure_query')
 
+            with super_group.group('property') as group:
```
You should add scenario tests for this, take a look at the existing tests to see how to write scenario tests and the [wiki documentation](https://github.com/Azure/service-fabric-cli/wiki/Testing)

jjcollinge replied:
I had a look at writing some scenario tests - the only Service Fabric name that I can guarantee is in the cluster is the 'System' name. I can test the 'list' and 'get' against it but it's restricted as far as 'put' and 'delete'. I can either deploy an application using the 'sfctl application' commands or update this PR to include a new property group for 'name' (i.e. sfctl name create "{\"Name\":\"TestName\"}") and couple the property tests to the name commands. How do you suggest I proceed? Docs on this API are here: https://docs.microsoft.com/en-us/rest/api/servicefabric/sfclient-index-property-management

samedder commented:
I forgot to mention in the review but please also update
- README.rst to include the new features
- Package version as this is a new API

samedder commented on src/setup.py:
```diff
@@ -17,7 +17,7 @@ def read(fname):
 setup(
     name='sfctl',
-    version='3.0.0',
+    version='3.0.1',
```
I'd say lets make this `3.1.0`

samedder commented:
@jjcollinge Looks good except for just the version. Do you have any tests/verification for this btw? I'm a bit weary checking this in without at least some minimal verification that the empty cases work correctly.

jjcollinge commented:
I updated the version. As it's a standard command there's no logic to unit test as such but I've added a few scenario tests. The only one I can positively validate on a 'clean' SF cluster is 'sfctl property list --name "System"' as it's the only name that exists. All other operations against "System" are restricted. The response for error conditions (i.e. referencing a non existent property or name) will just spit back the raw exception/stack trace with an error message such as (FabricErrorException: (FABRIC_E_PROPERTY_DOES_NOT_EXIST) Property does not exist). I would add logic to prettify this for the user but that's inconsistent with the rest of the sfctl commands.

samedder commented:
Thanks for the contribution @jjcollinge
